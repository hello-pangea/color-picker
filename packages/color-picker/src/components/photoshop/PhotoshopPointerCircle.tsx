import React from "react";
import { HslColor } from "../../types/colors";

type Props = {
  hsl: HslColor;
};

export default function PhotoshopPointerCircle({ hsl }: Props) {
  return (
    <div
      style={{
        width: "12px",
        height: "12px",
        borderRadius: "6px",
        boxShadow:
          hsl.l > 0.5 ? "inset 0 0 0 1px #000" : "inset 0 0 0 1px #fff",
        transform: "translate(-6px, -6px)",
      }}
    />
  );
}
