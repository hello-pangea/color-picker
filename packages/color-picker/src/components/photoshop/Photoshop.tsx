import merge from "lodash/merge";
import React, { useState } from "react";
import { useColor, withColorProvider } from "../../context/useColor";
import { Hue, Saturation } from "../common";
import PhotoshopButton from "./PhotoshopButton";
import PhotoshopFields from "./PhotoshopFields";
import PhotoshopPointer from "./PhotoshopPointer";
import PhotoshopPointerCircle from "./PhotoshopPointerCircle";
import PhotoshopPreviews from "./PhotoshopPreviews";

export type PhotoshopPickerProps = {
  header?: string;
  styles?: Record<string, React.CSSProperties>;
  className?: string;
  onAccept?: () => void;
  onCancel?: () => void;
};

function Photoshop({
  header = "Color Picker",
  styles: passedStyles = {},
  className = "",
  onAccept,
  onCancel,
}: PhotoshopPickerProps) {
  const { colors, changeColor } = useColor();
  const { rgb, hex, hsv, hsl } = colors;

  const [currentColor, setCurrentColor] = useState(hex);

  const styles = merge<
    Record<string, React.CSSProperties>,
    Record<string, React.CSSProperties>
  >(
    {
      picker: {
        background: "#DCDCDC",
        borderRadius: "4px",
        boxShadow: "0 0 0 1px rgba(0,0,0,.25), 0 8px 16px rgba(0,0,0,.15)",
        boxSizing: "initial",
        width: "513px",
      },
      head: {
        backgroundImage: "linear-gradient(-180deg, #F0F0F0 0%, #D4D4D4 100%)",
        borderBottom: "1px solid #B1B1B1",
        boxShadow:
          "inset 0 1px 0 0 rgba(255,255,255,.2), inset 0 -1px 0 0 rgba(0,0,0,.02)",
        height: "23px",
        lineHeight: "24px",
        borderRadius: "4px 4px 0 0",
        fontSize: "13px",
        color: "#4D4D4D",
        textAlign: "center",
      },
      body: {
        padding: "15px 15px 0",
        display: "flex",
      },
      saturation: {
        width: "256px",
        height: "256px",
        position: "relative",
        border: "2px solid #B3B3B3",
        borderBottom: "2px solid #F0F0F0",
        overflow: "hidden",
      },
      hue: {
        position: "relative",
        height: "256px",
        width: "19px",
        marginLeft: "10px",
        border: "2px solid #B3B3B3",
        borderBottom: "2px solid #F0F0F0",
      },
      controls: {
        width: "180px",
        marginLeft: "10px",
      },
      top: {
        display: "flex",
      },
      previews: {
        width: "60px",
      },
      actions: {
        flex: "1",
        marginLeft: "20px",
      },
    },
    passedStyles
  );

  function handleAccept() {
    onAccept && onAccept();
    setCurrentColor(hex);
  }

  return (
    <div style={styles.picker} className={`photoshop-picker ${className}`}>
      <div style={styles.head}>{header}</div>

      <div style={styles.body} className="flexbox-fix">
        <div style={styles.saturation}>
          <Saturation
            hsl={hsl}
            hsv={hsv}
            pointer={<PhotoshopPointerCircle hsl={hsl} />}
            onChange={changeColor}
          />
        </div>
        <div style={styles.hue}>
          <Hue
            direction="vertical"
            hsl={hsl}
            pointer={PhotoshopPointer}
            onChange={changeColor}
          />
        </div>
        <div style={styles.controls}>
          <div style={styles.top} className="flexbox-fix">
            <div style={styles.previews}>
              <PhotoshopPreviews rgb={rgb} currentColor={currentColor} />
            </div>
            <div style={styles.actions}>
              <PhotoshopButton label="OK" onClick={handleAccept} active />
              <PhotoshopButton label="Cancel" onClick={onCancel} />
              <PhotoshopFields
                onChange={changeColor}
                rgb={rgb}
                hsv={hsv}
                hex={hex}
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default withColorProvider(Photoshop);
