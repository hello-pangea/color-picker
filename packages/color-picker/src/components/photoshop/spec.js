/* global test, jest, expect */

import React from "react";
import renderer from "react-test-renderer";
import { red } from "../../helpers/color";

import Photoshop from "./Photoshop";
import PhotoshopButton from "./PhotoshopButton";
import PhotoshopFields from "./PhotoshopFields";
import PhotoshopPointer from "./PhotoshopPointer";
import PhotoshopPointerCircle from "./PhotoshopPointerCircle";
import PhotoshopPreviews from "./PhotoshopPreviews";

test("Photoshop renders correctly", () => {
  const tree = renderer
    .create(<Photoshop {...red} onAccept={() => {}} onCancel={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Photoshop renders custom styles correctly", () => {
  const tree = renderer
    .create(
      <Photoshop
        {...red}
        styles={{ default: { picker: { boxShadow: "0 0 10px red" } } }}
      />
    )
    .toJSON();
  expect(tree.props.style.boxShadow).toBe("0 0 10px red");
});

test("PhotoshopButton renders correctly", () => {
  const tree = renderer
    .create(<PhotoshopButton label="accept" onClick={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("PhotoshopFields renders correctly", () => {
  const tree = renderer.create(<PhotoshopFields {...red} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("PhotoshopPointer renders correctly", () => {
  const tree = renderer.create(<PhotoshopPointer />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("PhotoshopPointerCircle renders correctly", () => {
  const tree = renderer.create(<PhotoshopPointerCircle {...red} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("PhotoshopPreviews renders correctly", () => {
  const tree = renderer
    .create(<PhotoshopPreviews {...red} currencColor="#aeee00" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
