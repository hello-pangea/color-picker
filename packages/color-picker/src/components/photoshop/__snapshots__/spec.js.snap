// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photoshop renders correctly 1`] = `
<div
  className="photoshop-picker "
  style={
    {
      "background": "#DCDCDC",
      "borderRadius": "4px",
      "boxShadow": "0 0 0 1px rgba(0,0,0,.25), 0 8px 16px rgba(0,0,0,.15)",
      "boxSizing": "initial",
      "width": "513px",
    }
  }
>
  <div
    style={
      {
        "backgroundImage": "linear-gradient(-180deg, #F0F0F0 0%, #D4D4D4 100%)",
        "borderBottom": "1px solid #B1B1B1",
        "borderRadius": "4px 4px 0 0",
        "boxShadow": "inset 0 1px 0 0 rgba(255,255,255,.2), inset 0 -1px 0 0 rgba(0,0,0,.02)",
        "color": "#4D4D4D",
        "fontSize": "13px",
        "height": "23px",
        "lineHeight": "24px",
        "textAlign": "center",
      }
    }
  >
    Color Picker
  </div>
  <div
    className="flexbox-fix"
    style={
      {
        "display": "flex",
        "padding": "15px 15px 0",
      }
    }
  >
    <div
      style={
        {
          "border": "2px solid #B3B3B3",
          "borderBottom": "2px solid #F0F0F0",
          "height": "256px",
          "overflow": "hidden",
          "position": "relative",
          "width": "256px",
        }
      }
    >
      <div
        onMouseDown={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          {
            "background": "hsl(249.99999999999994,100%, 50%)",
            "borderRadius": undefined,
            "inset": "0px",
            "position": "absolute",
          }
        }
      >
        <style>
          
          .saturation-white {
            background: -webkit-linear-gradient(to right, #fff, rgba(255,255,255,0));
            background: linear-gradient(to right, #fff, rgba(255,255,255,0));
          }
          .saturation-black {
            background: -webkit-linear-gradient(to top, #000, rgba(0,0,0,0));
            background: linear-gradient(to top, #000, rgba(0,0,0,0));
          }
        
        </style>
        <div
          className="saturation-white"
          style={
            {
              "borderRadius": undefined,
              "inset": "0px",
              "position": "absolute",
            }
          }
        >
          <div
            className="saturation-black"
            style={
              {
                "borderRadius": undefined,
                "boxShadow": undefined,
                "inset": "0px",
                "position": "absolute",
              }
            }
          />
          <div
            style={
              {
                "cursor": "default",
                "left": "66.66666666666667%",
                "position": "absolute",
                "top": "70%",
              }
            }
          >
            <div
              style={
                {
                  "borderRadius": "6px",
                  "boxShadow": "inset 0 0 0 1px #fff",
                  "height": "12px",
                  "transform": "translate(-6px, -6px)",
                  "width": "12px",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        {
          "border": "2px solid #B3B3B3",
          "borderBottom": "2px solid #F0F0F0",
          "height": "256px",
          "marginLeft": "10px",
          "position": "relative",
          "width": "19px",
        }
      }
    >
      <div
        style={
          {
            "borderRadius": undefined,
            "boxShadow": undefined,
            "inset": "0px",
            "position": "absolute",
          }
        }
      >
        <div
          className="hue-vertical"
          onMouseDown={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "borderRadius": undefined,
              "height": "100%",
              "padding": "0 2px",
              "position": "relative",
            }
          }
        >
          <style>
            
            .hue-horizontal {
              background: linear-gradient(to right, #f00 0%, #ff0 17%, #0f0
                33%, #0ff 50%, #00f 67%, #f0f 83%, #f00 100%);
              background: -webkit-linear-gradient(to right, #f00 0%, #ff0
                17%, #0f0 33%, #0ff 50%, #00f 67%, #f0f 83%, #f00 100%);
            }

            .hue-vertical {
              background: linear-gradient(to top, #f00 0%, #ff0 17%, #0f0 33%,
                #0ff 50%, #00f 67%, #f0f 83%, #f00 100%);
              background: -webkit-linear-gradient(to top, #f00 0%, #ff0 17%,
                #0f0 33%, #0ff 50%, #00f 67%, #f0f 83%, #f00 100%);
            }
          
          </style>
          <div
            style={
              {
                "left": "0px",
                "position": "absolute",
                "top": "30.55555555555557%",
              }
            }
          >
            <div>
              <div
                style={
                  {
                    "borderColor": "transparent transparent transparent #555",
                    "borderStyle": "solid",
                    "borderWidth": "5px 0 5px 8px",
                    "height": 0,
                    "transform": "translate(-13px, -4px)",
                    "width": 0,
                  }
                }
              >
                <div
                  style={
                    {
                      "borderColor": "transparent transparent transparent #fff",
                      "borderStyle": "solid",
                      "borderWidth": "4px 0 4px 6px",
                      "height": 0,
                      "left": "1px",
                      "position": "absolute",
                      "top": "1px",
                      "transform": "translate(-8px, -5px)",
                      "width": 0,
                    }
                  }
                />
              </div>
              <div
                style={
                  {
                    "borderColor": "transparent transparent transparent #555",
                    "borderStyle": "solid",
                    "borderWidth": "5px 0 5px 8px",
                    "height": 0,
                    "transform": "translate(20px, -14px) rotate(180deg)",
                    "width": 0,
                  }
                }
              >
                <div
                  style={
                    {
                      "borderColor": "transparent transparent transparent #fff",
                      "borderStyle": "solid",
                      "borderWidth": "4px 0 4px 6px",
                      "height": 0,
                      "left": "1px",
                      "position": "absolute",
                      "top": "1px",
                      "transform": "translate(-8px, -5px)",
                      "width": 0,
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        {
          "marginLeft": "10px",
          "width": "180px",
        }
      }
    >
      <div
        className="flexbox-fix"
        style={
          {
            "display": "flex",
          }
        }
      >
        <div
          style={
            {
              "width": "60px",
            }
          }
        >
          <div>
            <div
              style={
                {
                  "color": "#000",
                  "fontSize": "14px",
                  "textAlign": "center",
                }
              }
            >
              new
            </div>
            <div
              style={
                {
                  "border": "1px solid #B3B3B3",
                  "borderBottom": "1px solid #F0F0F0",
                  "marginBottom": "2px",
                  "marginTop": "1px",
                }
              }
            >
              <div
                style={
                  {
                    "background": "rgb(34,25, 77)",
                    "boxShadow": "inset 1px 0 0 #000, inset -1px 0 0 #000, inset 0 1px 0 #000",
                    "height": "34px",
                  }
                }
              />
              <div
                style={
                  {
                    "background": "#22194d",
                    "boxShadow": "inset 1px 0 0 #000, inset -1px 0 0 #000, inset 0 -1px 0 #000",
                    "height": "34px",
                  }
                }
              />
            </div>
            <div
              style={
                {
                  "color": "#000",
                  "fontSize": "14px",
                  "textAlign": "center",
                }
              }
            >
              current
            </div>
          </div>
        </div>
        <div
          style={
            {
              "flex": "1",
              "marginLeft": "20px",
            }
          }
        >
          <div
            onClick={[Function]}
            style={
              {
                "backgroundImage": "linear-gradient(-180deg, #FFFFFF 0%, #E6E6E6 100%)",
                "border": "1px solid #878787",
                "borderRadius": "2px",
                "boxShadow": "0 0 0 1px #878787",
                "color": "#000",
                "cursor": "pointer",
                "fontSize": "14px",
                "height": "20px",
                "lineHeight": "20px",
                "marginBottom": "10px",
                "textAlign": "center",
              }
            }
          >
            OK
          </div>
          <div
            onClick={[Function]}
            style={
              {
                "backgroundImage": "linear-gradient(-180deg, #FFFFFF 0%, #E6E6E6 100%)",
                "border": "1px solid #878787",
                "borderRadius": "2px",
                "boxShadow": "0 1px 0 0 #EAEAEA",
                "color": "#000",
                "cursor": "pointer",
                "fontSize": "14px",
                "height": "20px",
                "lineHeight": "20px",
                "marginBottom": "10px",
                "textAlign": "center",
              }
            }
          >
            Cancel
          </div>
          <div
            style={
              {
                "paddingBottom": "9px",
                "paddingTop": "5px",
                "position": "relative",
                "width": "80px",
              }
            }
          >
            <div
              style={
                {
                  "position": "relative",
                }
              }
            >
              <input
                id="rc-editable-input-1"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                spellCheck="false"
                style={
                  {
                    "border": "1px solid #888888",
                    "boxShadow": "inset 0 1px 1px rgba(0,0,0,.1), 0 1px 0 0 #ECECEC",
                    "fontSize": "13px",
                    "height": "18px",
                    "marginBottom": "5px",
                    "marginLeft": "40%",
                    "marginRight": "10px",
                    "paddingLeft": "3px",
                    "width": "40%",
                  }
                }
                value="250"
              />
              <label
                htmlFor="rc-editable-input-1"
                onMouseDown={[Function]}
                style={
                  {
                    "cursor": undefined,
                    "fontSize": "13px",
                    "height": "18px",
                    "left": "0px",
                    "lineHeight": "22px",
                    "position": "absolute",
                    "textTransform": "uppercase",
                    "top": "0px",
                    "width": "34px",
                  }
                }
              >
                h
              </label>
            </div>
            <div
              style={
                {
                  "position": "relative",
                }
              }
            >
              <input
                id="rc-editable-input-2"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                spellCheck="false"
                style={
                  {
                    "border": "1px solid #888888",
                    "boxShadow": "inset 0 1px 1px rgba(0,0,0,.1), 0 1px 0 0 #ECECEC",
                    "fontSize": "13px",
                    "height": "18px",
                    "marginBottom": "5px",
                    "marginLeft": "40%",
                    "marginRight": "10px",
                    "paddingLeft": "3px",
                    "width": "40%",
                  }
                }
                value="67"
              />
              <label
                htmlFor="rc-editable-input-2"
                onMouseDown={[Function]}
                style={
                  {
                    "cursor": undefined,
                    "fontSize": "13px",
                    "height": "18px",
                    "left": "0px",
                    "lineHeight": "22px",
                    "position": "absolute",
                    "textTransform": "uppercase",
                    "top": "0px",
                    "width": "34px",
                  }
                }
              >
                s
              </label>
            </div>
            <div
              style={
                {
                  "position": "relative",
                }
              }
            >
              <input
                id="rc-editable-input-3"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                spellCheck="false"
                style={
                  {
                    "border": "1px solid #888888",
                    "boxShadow": "inset 0 1px 1px rgba(0,0,0,.1), 0 1px 0 0 #ECECEC",
                    "fontSize": "13px",
                    "height": "18px",
                    "marginBottom": "5px",
                    "marginLeft": "40%",
                    "marginRight": "10px",
                    "paddingLeft": "3px",
                    "width": "40%",
                  }
                }
                value="30"
              />
              <label
                htmlFor="rc-editable-input-3"
                onMouseDown={[Function]}
                style={
                  {
                    "cursor": undefined,
                    "fontSize": "13px",
                    "height": "18px",
                    "left": "0px",
                    "lineHeight": "22px",
                    "position": "absolute",
                    "textTransform": "uppercase",
                    "top": "0px",
                    "width": "34px",
                  }
                }
              >
                v
              </label>
            </div>
            <div
              style={
                {
                  "height": "5px",
                }
              }
            />
            <div
              style={
                {
                  "position": "relative",
                }
              }
            >
              <input
                id="rc-editable-input-4"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                spellCheck="false"
                style={
                  {
                    "border": "1px solid #888888",
                    "boxShadow": "inset 0 1px 1px rgba(0,0,0,.1), 0 1px 0 0 #ECECEC",
                    "fontSize": "13px",
                    "height": "18px",
                    "marginBottom": "5px",
                    "marginLeft": "40%",
                    "marginRight": "10px",
                    "paddingLeft": "3px",
                    "width": "40%",
                  }
                }
                value="34"
              />
              <label
                htmlFor="rc-editable-input-4"
                onMouseDown={[Function]}
                style={
                  {
                    "cursor": undefined,
                    "fontSize": "13px",
                    "height": "18px",
                    "left": "0px",
                    "lineHeight": "22px",
                    "position": "absolute",
                    "textTransform": "uppercase",
                    "top": "0px",
                    "width": "34px",
                  }
                }
              >
                r
              </label>
            </div>
            <div
              style={
                {
                  "position": "relative",
                }
              }
            >
              <input
                id="rc-editable-input-5"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                spellCheck="false"
                style={
                  {
                    "border": "1px solid #888888",
                    "boxShadow": "inset 0 1px 1px rgba(0,0,0,.1), 0 1px 0 0 #ECECEC",
                    "fontSize": "13px",
                    "height": "18px",
                    "marginBottom": "5px",
                    "marginLeft": "40%",
                    "marginRight": "10px",
                    "paddingLeft": "3px",
                    "width": "40%",
                  }
                }
                value="25"
              />
              <label
                htmlFor="rc-editable-input-5"
                onMouseDown={[Function]}
                style={
                  {
                    "cursor": undefined,
                    "fontSize": "13px",
                    "height": "18px",
                    "left": "0px",
                    "lineHeight": "22px",
                    "position": "absolute",
                    "textTransform": "uppercase",
                    "top": "0px",
                    "width": "34px",
                  }
                }
              >
                g
              </label>
            </div>
            <div
              style={
                {
                  "position": "relative",
                }
              }
            >
              <input
                id="rc-editable-input-6"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                spellCheck="false"
                style={
                  {
                    "border": "1px solid #888888",
                    "boxShadow": "inset 0 1px 1px rgba(0,0,0,.1), 0 1px 0 0 #ECECEC",
                    "fontSize": "13px",
                    "height": "18px",
                    "marginBottom": "5px",
                    "marginLeft": "40%",
                    "marginRight": "10px",
                    "paddingLeft": "3px",
                    "width": "40%",
                  }
                }
                value="77"
              />
              <label
                htmlFor="rc-editable-input-6"
                onMouseDown={[Function]}
                style={
                  {
                    "cursor": undefined,
                    "fontSize": "13px",
                    "height": "18px",
                    "left": "0px",
                    "lineHeight": "22px",
                    "position": "absolute",
                    "textTransform": "uppercase",
                    "top": "0px",
                    "width": "34px",
                  }
                }
              >
                b
              </label>
            </div>
            <div
              style={
                {
                  "height": "5px",
                }
              }
            />
            <div
              style={
                {
                  "position": "relative",
                }
              }
            >
              <input
                id="rc-editable-input-7"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                spellCheck="false"
                style={
                  {
                    "border": "1px solid #888888",
                    "boxShadow": "inset 0 1px 1px rgba(0,0,0,.1), 0 1px 0 0 #ECECEC",
                    "fontSize": "13px",
                    "height": "18px",
                    "marginBottom": "6px",
                    "marginLeft": "20%",
                    "paddingLeft": "3px",
                    "width": "80%",
                  }
                }
                value="22194D"
              />
              <label
                htmlFor="rc-editable-input-7"
                onMouseDown={[Function]}
                style={
                  {
                    "cursor": undefined,
                    "fontSize": "13px",
                    "height": "18px",
                    "left": "0px",
                    "lineHeight": "22px",
                    "position": "absolute",
                    "textTransform": "uppercase",
                    "top": "0px",
                    "width": "14px",
                  }
                }
              >
                #
              </label>
            </div>
            <div
              style={
                {
                  "fontSize": "13px",
                  "position": "absolute",
                  "right": "-7px",
                  "top": "5px",
                }
              }
            >
              <div
                style={
                  {
                    "height": "20px",
                    "lineHeight": "22px",
                    "paddingBottom": "7px",
                  }
                }
              >
                °
              </div>
              <div
                style={
                  {
                    "height": "20px",
                    "lineHeight": "22px",
                    "paddingBottom": "7px",
                  }
                }
              >
                %
              </div>
              <div
                style={
                  {
                    "height": "20px",
                    "lineHeight": "22px",
                    "paddingBottom": "7px",
                  }
                }
              >
                %
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PhotoshopButton renders correctly 1`] = `
<div
  onClick={[Function]}
  style={
    {
      "backgroundImage": "linear-gradient(-180deg, #FFFFFF 0%, #E6E6E6 100%)",
      "border": "1px solid #878787",
      "borderRadius": "2px",
      "boxShadow": "0 1px 0 0 #EAEAEA",
      "color": "#000",
      "cursor": "pointer",
      "fontSize": "14px",
      "height": "20px",
      "lineHeight": "20px",
      "marginBottom": "10px",
      "textAlign": "center",
    }
  }
>
  accept
</div>
`;

exports[`PhotoshopFields renders correctly 1`] = `
<div
  style={
    {
      "paddingBottom": "9px",
      "paddingTop": "5px",
      "position": "relative",
      "width": "80px",
    }
  }
>
  <div
    style={
      {
        "position": "relative",
      }
    }
  >
    <input
      id="rc-editable-input-15"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      spellCheck="false"
      style={
        {
          "border": "1px solid #888888",
          "boxShadow": "inset 0 1px 1px rgba(0,0,0,.1), 0 1px 0 0 #ECECEC",
          "fontSize": "13px",
          "height": "18px",
          "marginBottom": "5px",
          "marginLeft": "40%",
          "marginRight": "10px",
          "paddingLeft": "3px",
          "width": "40%",
        }
      }
      value="0"
    />
    <label
      htmlFor="rc-editable-input-15"
      onMouseDown={[Function]}
      style={
        {
          "cursor": undefined,
          "fontSize": "13px",
          "height": "18px",
          "left": "0px",
          "lineHeight": "22px",
          "position": "absolute",
          "textTransform": "uppercase",
          "top": "0px",
          "width": "34px",
        }
      }
    >
      h
    </label>
  </div>
  <div
    style={
      {
        "position": "relative",
      }
    }
  >
    <input
      id="rc-editable-input-16"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      spellCheck="false"
      style={
        {
          "border": "1px solid #888888",
          "boxShadow": "inset 0 1px 1px rgba(0,0,0,.1), 0 1px 0 0 #ECECEC",
          "fontSize": "13px",
          "height": "18px",
          "marginBottom": "5px",
          "marginLeft": "40%",
          "marginRight": "10px",
          "paddingLeft": "3px",
          "width": "40%",
        }
      }
      value="100"
    />
    <label
      htmlFor="rc-editable-input-16"
      onMouseDown={[Function]}
      style={
        {
          "cursor": undefined,
          "fontSize": "13px",
          "height": "18px",
          "left": "0px",
          "lineHeight": "22px",
          "position": "absolute",
          "textTransform": "uppercase",
          "top": "0px",
          "width": "34px",
        }
      }
    >
      s
    </label>
  </div>
  <div
    style={
      {
        "position": "relative",
      }
    }
  >
    <input
      id="rc-editable-input-17"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      spellCheck="false"
      style={
        {
          "border": "1px solid #888888",
          "boxShadow": "inset 0 1px 1px rgba(0,0,0,.1), 0 1px 0 0 #ECECEC",
          "fontSize": "13px",
          "height": "18px",
          "marginBottom": "5px",
          "marginLeft": "40%",
          "marginRight": "10px",
          "paddingLeft": "3px",
          "width": "40%",
        }
      }
      value="100"
    />
    <label
      htmlFor="rc-editable-input-17"
      onMouseDown={[Function]}
      style={
        {
          "cursor": undefined,
          "fontSize": "13px",
          "height": "18px",
          "left": "0px",
          "lineHeight": "22px",
          "position": "absolute",
          "textTransform": "uppercase",
          "top": "0px",
          "width": "34px",
        }
      }
    >
      v
    </label>
  </div>
  <div
    style={
      {
        "height": "5px",
      }
    }
  />
  <div
    style={
      {
        "position": "relative",
      }
    }
  >
    <input
      id="rc-editable-input-18"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      spellCheck="false"
      style={
        {
          "border": "1px solid #888888",
          "boxShadow": "inset 0 1px 1px rgba(0,0,0,.1), 0 1px 0 0 #ECECEC",
          "fontSize": "13px",
          "height": "18px",
          "marginBottom": "5px",
          "marginLeft": "40%",
          "marginRight": "10px",
          "paddingLeft": "3px",
          "width": "40%",
        }
      }
      value="255"
    />
    <label
      htmlFor="rc-editable-input-18"
      onMouseDown={[Function]}
      style={
        {
          "cursor": undefined,
          "fontSize": "13px",
          "height": "18px",
          "left": "0px",
          "lineHeight": "22px",
          "position": "absolute",
          "textTransform": "uppercase",
          "top": "0px",
          "width": "34px",
        }
      }
    >
      r
    </label>
  </div>
  <div
    style={
      {
        "position": "relative",
      }
    }
  >
    <input
      id="rc-editable-input-19"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      spellCheck="false"
      style={
        {
          "border": "1px solid #888888",
          "boxShadow": "inset 0 1px 1px rgba(0,0,0,.1), 0 1px 0 0 #ECECEC",
          "fontSize": "13px",
          "height": "18px",
          "marginBottom": "5px",
          "marginLeft": "40%",
          "marginRight": "10px",
          "paddingLeft": "3px",
          "width": "40%",
        }
      }
      value="0"
    />
    <label
      htmlFor="rc-editable-input-19"
      onMouseDown={[Function]}
      style={
        {
          "cursor": undefined,
          "fontSize": "13px",
          "height": "18px",
          "left": "0px",
          "lineHeight": "22px",
          "position": "absolute",
          "textTransform": "uppercase",
          "top": "0px",
          "width": "34px",
        }
      }
    >
      g
    </label>
  </div>
  <div
    style={
      {
        "position": "relative",
      }
    }
  >
    <input
      id="rc-editable-input-20"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      spellCheck="false"
      style={
        {
          "border": "1px solid #888888",
          "boxShadow": "inset 0 1px 1px rgba(0,0,0,.1), 0 1px 0 0 #ECECEC",
          "fontSize": "13px",
          "height": "18px",
          "marginBottom": "5px",
          "marginLeft": "40%",
          "marginRight": "10px",
          "paddingLeft": "3px",
          "width": "40%",
        }
      }
      value="0"
    />
    <label
      htmlFor="rc-editable-input-20"
      onMouseDown={[Function]}
      style={
        {
          "cursor": undefined,
          "fontSize": "13px",
          "height": "18px",
          "left": "0px",
          "lineHeight": "22px",
          "position": "absolute",
          "textTransform": "uppercase",
          "top": "0px",
          "width": "34px",
        }
      }
    >
      b
    </label>
  </div>
  <div
    style={
      {
        "height": "5px",
      }
    }
  />
  <div
    style={
      {
        "position": "relative",
      }
    }
  >
    <input
      id="rc-editable-input-21"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      spellCheck="false"
      style={
        {
          "border": "1px solid #888888",
          "boxShadow": "inset 0 1px 1px rgba(0,0,0,.1), 0 1px 0 0 #ECECEC",
          "fontSize": "13px",
          "height": "18px",
          "marginBottom": "6px",
          "marginLeft": "20%",
          "paddingLeft": "3px",
          "width": "80%",
        }
      }
      value="FF0000"
    />
    <label
      htmlFor="rc-editable-input-21"
      onMouseDown={[Function]}
      style={
        {
          "cursor": undefined,
          "fontSize": "13px",
          "height": "18px",
          "left": "0px",
          "lineHeight": "22px",
          "position": "absolute",
          "textTransform": "uppercase",
          "top": "0px",
          "width": "14px",
        }
      }
    >
      #
    </label>
  </div>
  <div
    style={
      {
        "fontSize": "13px",
        "position": "absolute",
        "right": "-7px",
        "top": "5px",
      }
    }
  >
    <div
      style={
        {
          "height": "20px",
          "lineHeight": "22px",
          "paddingBottom": "7px",
        }
      }
    >
      °
    </div>
    <div
      style={
        {
          "height": "20px",
          "lineHeight": "22px",
          "paddingBottom": "7px",
        }
      }
    >
      %
    </div>
    <div
      style={
        {
          "height": "20px",
          "lineHeight": "22px",
          "paddingBottom": "7px",
        }
      }
    >
      %
    </div>
  </div>
</div>
`;

exports[`PhotoshopPointer renders correctly 1`] = `
<div>
  <div
    style={
      {
        "borderColor": "transparent transparent transparent #555",
        "borderStyle": "solid",
        "borderWidth": "5px 0 5px 8px",
        "height": 0,
        "transform": "translate(-13px, -4px)",
        "width": 0,
      }
    }
  >
    <div
      style={
        {
          "borderColor": "transparent transparent transparent #fff",
          "borderStyle": "solid",
          "borderWidth": "4px 0 4px 6px",
          "height": 0,
          "left": "1px",
          "position": "absolute",
          "top": "1px",
          "transform": "translate(-8px, -5px)",
          "width": 0,
        }
      }
    />
  </div>
  <div
    style={
      {
        "borderColor": "transparent transparent transparent #555",
        "borderStyle": "solid",
        "borderWidth": "5px 0 5px 8px",
        "height": 0,
        "transform": "translate(20px, -14px) rotate(180deg)",
        "width": 0,
      }
    }
  >
    <div
      style={
        {
          "borderColor": "transparent transparent transparent #fff",
          "borderStyle": "solid",
          "borderWidth": "4px 0 4px 6px",
          "height": 0,
          "left": "1px",
          "position": "absolute",
          "top": "1px",
          "transform": "translate(-8px, -5px)",
          "width": 0,
        }
      }
    />
  </div>
</div>
`;

exports[`PhotoshopPointerCircle renders correctly 1`] = `
<div
  style={
    {
      "borderRadius": "6px",
      "boxShadow": "inset 0 0 0 1px #fff",
      "height": "12px",
      "transform": "translate(-6px, -6px)",
      "width": "12px",
    }
  }
/>
`;

exports[`PhotoshopPreviews renders correctly 1`] = `
<div>
  <div
    style={
      {
        "color": "#000",
        "fontSize": "14px",
        "textAlign": "center",
      }
    }
  >
    new
  </div>
  <div
    style={
      {
        "border": "1px solid #B3B3B3",
        "borderBottom": "1px solid #F0F0F0",
        "marginBottom": "2px",
        "marginTop": "1px",
      }
    }
  >
    <div
      style={
        {
          "background": "rgb(255,0, 0)",
          "boxShadow": "inset 1px 0 0 #000, inset -1px 0 0 #000, inset 0 1px 0 #000",
          "height": "34px",
        }
      }
    />
    <div
      style={
        {
          "background": undefined,
          "boxShadow": "inset 1px 0 0 #000, inset -1px 0 0 #000, inset 0 -1px 0 #000",
          "height": "34px",
        }
      }
    />
  </div>
  <div
    style={
      {
        "color": "#000",
        "fontSize": "14px",
        "textAlign": "center",
      }
    }
  >
    current
  </div>
</div>
`;
